src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -886,6 +889,16 @@ private void HandleProcessExit(object sender, EventArgs e)
             }
         }
 
+        private void DisposeLogProviders(IServiceProvider svc)
+        {
+            var providers = svc.GetServices<ILoggerProvider>();
```

Comment:
This shouldn't be required - the container is disposed immediately after this call and it is responsible for disposing these.

Reply:
no it isn't disposing the providers. I tested it.

Reply:
nvm. you are right. I was doing things wrong